proto/p4/p4runtime.proto

```diff
@@ -270,24 +281,31 @@ message MeterConfig {
 //  - DELETE: Delete the counter cell (i.e. reset the counter data)
 // For ReadRequest, the scope is defined as follows:
 // - All counter cells for all meters if counter_id = 0 (default).
-// - Indirect: All counter cells for given counter_id if index = 0 (default).
-// - Direct: All counter cells for given counter_id if table_entry.match is
-//   empty.
+// - All counter cells for given counter_id if index = 0 (default).
 message CounterEntry {
   uint32 counter_id = 1;
-  oneof type {
-    // For indirect counter
-    int64 index = 2;
-    // For direct counter, the associated table entry match.
-    // table_entry.action is ignored. Other fields specify the match.
-    TableEntry table_entry = 3;
-  }
-  CounterData data = 4;
-
+  int64 index = 2;
+  CounterData data = 3;
   // TODO(wmohsin) Remove once partners have moved to new API.
   repeated CounterCell cells = 5;
```

Comment:
Sorry, I'm learning the github code review UI :) I thought my comment wouldn't get published out until I PTAL at the end.
By the way, how about removing the Counter{Read|Write} RPCs as well (if you think CounterCell can be removed now)?